rust/src/compose.rs

```diff
@@ -113,11 +118,43 @@ fn fetch_previous_metadata(
         .ok_or_else(|| anyhow!("Missing config label {INPUTHASH_KEY}"))?
         .to_owned();
     Ok(ImageMetadata {
+        manifest,
         version: labels.get("version").map(ToOwned::to_owned),
         inputhash,
     })
 }
 
+#[derive(Debug, Default)]
+struct ManifestDiff {
+    removed: Vec<oci_spec::image::Descriptor>,
+    added: Vec<oci_spec::image::Descriptor>,
+}
+
+fn manifest_diff(src: &ImageManifest, dest: &ImageManifest) -> ManifestDiff {
+    let src_layers = src
+        .layers()
+        .iter()
+        .map(|l| (l.digest(), l))
+        .collect::<HashMap<_, _>>();
+    let dest_layers = dest
+        .layers()
+        .iter()
+        .map(|l| (l.digest(), l))
+        .collect::<HashMap<_, _>>();
```

Comment:
Minor: as you are iterating on this (and the one above), I think a `BTreeMap` would suit better.